### PR TITLE
Un-revert add /tmp emptyDir volume to connector pods

### PR DESCRIPTION
**Pull request by pmossman**
## What
Addresses https://github.com/airbytehq/airbyte/issues/11163

Context: in https://github.com/airbytehq/airbyte/pull/10761, a kube volume mount at `/tmp` was added for connector pods that need to write temporary data to disk. This was reverted because kube acceptance tests were consistently failing after this change. 

After some digging, I found that normalization was failing due to an empty `/tmp/dbt_modules` directory, and this made sense as the new volume mount at `/tmp` was clearly overwriting the compile-time dbt modules that were stored there.

Slack thread with some context/convo: https://airbytehq-team.slack.com/archives/C02TXQ020QM/p1648763773988619

## How
This PR updates our dbt project templates to store dbt_modules at `/dbt-tmp` instead of `/tmp` so that the volume mount no longer conflicts. It also un-reverts the original PR to re-introduce the `/tmp` volume mount.

## Misc Questions
- Is this the correct way to make a change to `base-normalization`? I'll need a `/publish` command at some point right? (I haven't ever had to use `publish` so this is the first time I'm running into this process).

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/dbt-project-template-clickhouse/dbt_project.yml
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-normalization/dbt-project-template-clickhouse/dbt_project.yml:27]: for simpler naming, it doesn't need to be related to `tmp`, what about going directly with `/dbt_modules` or even `/dbt` (without the dbt_modules)?

**Comment by ChristopheDuong** [comment]
> Is this the correct way to make a change to base-normalization? I'll need a /publish command at some point right? 

Yes, you have description in the release process doc here too: https://docs.google.com/document/d/1Q1N4oomtz3rXqqn97nMB5rfcM_Dd2K3in5vmYPWdjjY/edit

File: airbyte-integrations/bases/base-normalization/dbt-project-template-clickhouse/dbt_project.yml
Comment (davinchia) [reply airbyte-integrations/bases/base-normalization/dbt-project-template-clickhouse/dbt_project.yml:27]: I like `/dbt` too. It's not really tmp since these are files we want to keep and continually use.

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mssql/test_simple_streams/dbt_project.yml
Comment (davinchia) [inline airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mssql/test_simple_streams/dbt_project.yml:35]: curious, what's doing this format?

**Review by davinchia — APPROVED** [review]
LGTM! What Chris said about testing and publishing before we merge this in.

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mssql/test_simple_streams/dbt_project.yml
Comment (pmossman) [reply airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mssql/test_simple_streams/dbt_project.yml:35]: I believe this was the 'prettier' pre-commit hook

**Comment by pmossman** [comment]
@ChristopheDuong I tried to run normalization integration tests locally, but I wasn't able to get this working on my M1 laptop. I don't think I ever managed to get any of the `SUB_BUILD=CONNECTORS_BASE` gradle workflows working locally.

Is there a workaround to generate those `dbt_project.yml` changes you mentioned? Or alternatively, would you be able to check out this branch and run the integration tests locally on your machine?

**Comment by pmossman** [comment]
Actually ignore my last comment @ChristopheDuong , I asked Edward for M1 python help and we resolved the blocker I was running into, running the local integration tests now and will follow the release process guidelines now that I'm unblocked

**Comment by pmossman** [comment]
Sorry for the message spam @ChristopheDuong but I spoke too soon I think - normalization integration tests are failing for me, even if I run in master. I don't want to block this PR on my M1-specific woes so if you or @davinchia  able to run the integration tests locally for this branch and commit the generated file changes, that'd be greatly appreciated!

**Comment by ChristopheDuong** [comment]
it seems like some .sql files for clickhouse/mssql were not up to date either, so it's now part of your PR.

I also had to add a little change in `airbyte-integrations/bases/base-normalization/integration_tests/dbt_integration_test.py` too

**Comment by pmossman** [comment]
Thanks so much Chris!

**Comment by edgao** [comment]
I'm not yet sure what's going on with the integration tests, but I ran into the same error (`function json_extract_path_text(super, "unknown", boolean) does not exist`) on another PR (https://github.com/airbytehq/airbyte/pull/11804). Trying to repro on master right now (https://github.com/airbytehq/airbyte/runs/5873579710?check_suite_focus=true).

**Comment by ChristopheDuong** [comment]
You can talk to @girarda, he ran into that weeks ago too and had to solve it already i think.

IIRC this is because this PR has side effects on all other branches: https://github.com/airbytehq/airbyte/pull/9610
This is due to the datatypes of the tests being changed to `super` datatype on the destination whereas your PRs are not handling that type change...

**Comment by edgao** [comment]
@Phlair It looks like the last publish run failed in "Check if connector definition is in definitions.yaml" - I'm guessing this is because normalization isn't actually tracked in there? I.e. this PR should be ready to merge without further intervention?

**Comment by Phlair** [comment]
@edgao correct, no auto-bump for normalization (yet). This makes me think the automated comments could be clearer since this shouldn't __look like__ a fail.

**Review by edgao — APPROVED** [review]
@pmossman I added an entry to the changelog; if that looks right to you then we should be good to merge here!

**Comment by pmossman** [comment]
Thanks so much @edgao for all your help getting this over the finish line! Changelog looks good, going to merge this now

**Pull request merged**
